packages/dds/tree/src/simple-tree/arrayNode.ts

```diff
@@ -620,8 +624,8 @@ function createArrayNodeProxy(
 				return Reflect.set(dispatch, key, newValue, proxy);
 			}
 
-			const field = getSequenceField(proxy);
-			const maybeIndex = asIndex(key, field.length);
+			// JS allows arrays to be set an any index, even an index that is arbitrarily past what is currently the end of the array.
+			const maybeIndex = asIndex(key, Infinity);
```

Comment:
While in modern browsers it doesn't matter, but I think Number.POSITIVE_INFINITY is preferred since getting your constants from number works better (has more useful constants) than pulling it from the global window object.
```suggestion
			const maybeIndex = asIndex(key, Number.POSITIVE_INFINITY);
```

Reply:
Interesting, I'll update.